Hm all the tests seem to fail with the same error:
```
musescore> running tests
musescore> check flags: -j16 test
musescore> [0/1] Running tests...
musescore> Test project /build/source/build
musescore>       Start  1: global_tests
musescore>       Start  2: accessibility_tests
musescore>       Start  3: audio_test
musescore>       Start  4: draw_tests
musescore>       Start  5: mpe_test
musescore>       Start  6: ui_tests
musescore>       Start  7: braille_tests
musescore>       Start  8: diagnostics_tests
musescore>       Start  9: engraving_tests
musescore>       Start 10: iex_bb_tests
musescore>       Start 11: iex_bww_tests
musescore>       Start 12: iex_capella_tests
musescore>       Start 13: iex_midi_tests
musescore>       Start 14: iex_musicxml_tests
musescore>       Start 15: iex_guitarpro_tests
musescore>       Start 16: plugins_tests
musescore>  1/17 Test #15: iex_guitarpro_tests ..............Subprocess aborted***Exception:   0.11 sec
musescore> qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
musescore> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
musescore> Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.
musescore>       Start 17: project_test
musescore>  2/17 Test  #3: audio_test .......................Subprocess aborted***Exception:   0.12 sec
musescore> qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
musescore> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
musescore> Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.

...
musescore> 16/17 Test  #9: engraving_tests ..................Subprocess aborted***Exception:   0.13 sec
musescore> qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
musescore> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
musescore> Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.
musescore> 17/17 Test #17: project_test .....................Subprocess aborted***Exception:   0.22 sec
musescore> qt.qpa.plugin: Could not find the Qt platform plugin "xcb" in ""
musescore> This application failed to start because no Qt platform plugin could be initialized. Reinstalling the application may fix this problem.
musescore> Available platform plugins are: wayland-egl, wayland, wayland-xcomposite-egl, wayland-xcomposite-glx.
musescore> 0% tests passed, 17 tests failed out of 17
musescore> Total Test time (real) =   0.35 sec
musescore> The following tests FAILED:
musescore>        1 - global_tests (Subprocess aborted)
musescore>        2 - accessibility_tests (Subprocess aborted)
musescore>        3 - audio_test (Subprocess aborted)
musescore>        4 - draw_tests (Subprocess aborted)
musescore>        5 - mpe_test (Subprocess aborted)
musescore>        6 - ui_tests (Subprocess aborted)
musescore>        7 - braille_tests (Subprocess aborted)
musescore>        8 - diagnostics_tests (Subprocess aborted)
musescore>        9 - engraving_tests (Subprocess aborted)
musescore>       10 - iex_bb_tests (Subprocess aborted)
musescore>       11 - iex_bww_tests (Subprocess aborted)
musescore>       12 - iex_capella_tests (Subprocess aborted)
musescore>       13 - iex_midi_tests (Subprocess aborted)
musescore>       14 - iex_musicxml_tests (Subprocess aborted)
musescore>       15 - iex_guitarpro_tests (Subprocess aborted)
musescore>       16 - plugins_tests (Subprocess aborted)
musescore>       17 - project_test (Subprocess aborted)
musescore> Errors while running CTest
musescore> FAILED: CMakeFiles/test.util
musescore> cd /build/source/build && /nix/store/p6nyqsw5qbg114gprcj4ci8j4nx9nhnb-cmake-3.26.4/bin/ctest --force-new-ctest-process
musescore> ninja: build stopped: subcommand failed.
```

Not sure if these are all the tests there are to run, given https://github.com/musescore/MuseScore/blob/efd683db329e2588a04d1828a2c3b140d0a25a55/CMakeLists.txt#L48-L49

--- a/pkgs/applications/audio/musescore/default.nix
+++ b/pkgs/applications/audio/musescore/default.nix
@@ -153,6 +153,9 @@ in stdenv'.mkDerivation rec {
     ln -s $out/Applications/mscore.app/Contents/MacOS/mscore $out/bin/mscore.
   '';
 
+  # Don't run bundled upstreams tests, as they require a running X window system.
+  doCheck = false;
+
   passthru.tests = nixosTests.musescore;
 
   meta = with lib; {

